### PR TITLE
Extend search ticket for support side load

### PR DESCRIPTION
**Pull request by rosa-soto**
## Description
Currently searchTickets returns just the group id, this PR extends the method for:
- Receive a sideLoadOption If side load is null searchMethod shouldn't set include parameter and should return the same response as currently is.
-  Make the call to zendesk api using "include" as parameter in order to retrieve the group object as a side load. (https://developer.zendesk.com/api-reference/ticketing/ticket-management/search/#side-loading)
The changes will be reflected in the UI by displaying the missing group name
Screen Shot 2023-04-12 at 8 40 32 AM

Screen Shot 2023-04-12 at 1 38 33 PM

@vendasta/warped-tour 
[WARP-33](https://vendasta.jira.com/browse/WARP-33)

[WARP-33]: https://vendasta.jira.com/browse/WARP-33?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: zendesk/search.go [hunk 1]
```diff
@@ -68,7 +68,8 @@ func OrganizationFilter(organizationID int) Filters {
 // SearchTickets leverages the unified search api to return tickets
 //
 // Zendesk Core API docs: https://developer.zendesk.com/rest_api/docs/support/search
-func (c *client) SearchTickets(term string, options *ListOptions, filters ...Filters) (*TicketSearchResults, error) {
+// Zendesk Side load API docs: https://developer.zendesk.com/api-reference/ticketing/ticket-management/search/#side-loading
+func (c *client) SearchTickets(term string, options *ListOptions, sideload SideLoad, filters ...Filters) (*TicketSearchResults, error) {
```
Comment (taylorwiebe) [inline zendesk/search.go]: should "SideLoad" be an array? it seems like you could have multiple options.

File: zendesk/search.go [hunk 2]
```diff
@@ -83,6 +84,14 @@ func (c *client) SearchTickets(term string, options *ListOptions, filters ...Fil
 		queryString = fmt.Sprintf(`%s /"%s/"`, queryString, term)
 	}
 	params.Set("query", queryString)
+	if sideload != nil {
+		sideLoads := &SideLoadOptions{}
+		sideload(sideLoads)
+		if len(sideLoads.Include) > 0 {
+			params.Set("include", fmt.Sprintf("tickets(%s)", strings.Join(sideLoads.Include, ",")))
+		}
+	}
```
Comment (taylorwiebe) [inline zendesk/search.go]: if you make side load an array you should be able to change this to:

```
	sideLoads := &SideLoadOptions{}
	for _, opt := range sideloads {
		opt(sideLoads)
	}
	if len(sideLoads.Include) > 0 {
		params.Set("include", strings.Join(sideLoads.Include, ","))
	}
```

File: zendesk/search_test.go [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+import (
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestSearchTickets(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test in short mode.")
+	}
```
Comment (benjfactor) [inline zendesk/search_test.go:11]: Do these tests take a long time?

Comment (rosa-soto) [reply zendesk/search_test.go:11]: No, it was fast.. but all the test in this wrapper have that option so I added there

File: zendesk/search.go (hunk 1 above)
Comment (rosa-soto) [reply zendesk/search.go]: the first time I did it was giving me errors but now is working well with an array.. I think was my mistake when testing this code. Added the support for arrays in sideload

**Review by bhingston-va — APPROVED** [review]

**Pull request merged**